Xamarin/DrawXShared/RealmDraw.cs

```diff
@@ -0,0 +1,552 @@
+        #endregion
+
+        #region Settings
+        private DrawXSettings Settings => DrawXSettingsManager.Settings;
+
+        private int _currentColorIndex = -1;  //// for quick check if pencil we draw is current color
+
+        private SwatchColor _currentColor;
+
+        private SwatchColor CurrentColor
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_currentColor.Name))
+                {
+                    InitCurrentColor();
+                }
+
+                return _currentColor;
+            }
+
+            set
+            {
+                if (!_currentColor.Name.Equals(value.Name))
+                {
+                    _currentColor = value;
+                    DrawXSettingsManager.Write(() => Settings.LastColorUsed = _currentColor.Name);
+                    _currentColorIndex = SwatchColor.Colors.IndexOf(_currentColor);
+                }
+            }
+        }
+
+        private void InitCurrentColor()
+        {
+            _currentColor = SwatchColor.ColorsByName [Settings.LastColorUsed];
+            _currentColorIndex = SwatchColor.Colors.IndexOf(_currentColor);
+        }
+        #endregion Settings
+
+        public RealmDraw(float inWidth, float inHeight)
+        {
+            // TODO close the Realm
+            _canvasWidth = inWidth;
+            _canvasHeight = inHeight;
+
+            // simple local open            
+            // _realm = Realm.GetInstance("DrawX.realm");
+
+            _pencilBitmaps = new List<SKBitmap>(SwatchColor.Colors.Count);
+            foreach (var swatch in SwatchColor.Colors)
+            {
+                _pencilBitmaps.Add(EmbeddedMedia.BitmapNamed(swatch.Name + ".png"));
+            }
+
+            _loginIconBitmap = EmbeddedMedia.BitmapNamed("CloudIcon.png");
+        }
+
+        internal void InvalidateCachedPaths()
+        {
+            _redrawPathsAtNextDraw = true;
+            _hasSavedBitmap = false;
+            _currentlyDrawing = null;
+        }
+
+        internal async void LoginToServerAsync()
+        {
+            // in case have lingering subscriptions, clear by clearing the results to which we subscribe
+            _allPaths = null;
+
+            _waitingForLogin = true;
+            var s = Settings;
+            //// TODO allow entering Create User flag on credentials to pass in here instead of false
+            var credentials = Credentials.UsernamePassword(s.Username, s.Password, false);
+            var user = await User.LoginAsync(credentials, new Uri($"http://{s.ServerIP}"));
+            Debug.WriteLine($"Got user logged in with refresh token {user.RefreshToken}");
+
+            var loginConf = new SyncConfiguration(user, new Uri($"realm://{s.ServerIP}/~/Draw"));
+            _realm = Realm.GetInstance(loginConf);
+            _allPaths = _realm.All<DrawPath>() as IQueryable<DrawPath>;
+            _allPaths.SubscribeForNotifications((sender, changes, error) =>
```

Comment:
This lambda should probably be factored out as it doesn't seem related to logging in..